### PR TITLE
Extra module for utility functions

### PR DESCRIPTION
**Pull request by swistakm**
Refers to #33.

This creates a new extension named `imgui.extra` that contains all additional utility functions that are not part of core C++ API. It also leaves aliases with deprecation warnings for old function names in `imgui.core` extension that will re removed in 1.0.0 version.

This unfortunately increases the project complexity a bit but it is possible that net complexity will decrease over time as we start to add new helpers (maybe even custom python-specific widgets!) and after we remove the deprecated aliases.

@pconerly would you like take a look and review it?

### CODE REVIEW

**Comment by swistakm** [comment]
I have just realized that this way of doing this (another Cython extension) is just a dead end. Both extensions will be statically linked against ImGui and this will break the whole integration of C++ assert statements with Python exceptions. Also this may break in unexpected ways.

Right now I so see only following solutions to this isse:
* `imgui.extra` module will be pure Python submodule that normally imports `imgui.core` module and calls its functions. This is the simplest and cleanest. We may lose the potential performance benefit of Cython cdefs but make whole packaging simpler. Still, it is possible that performance impact will be negligible.
*  We can leave all extra utilities in `imgui.core` module and "shadow" them using simple prefix (e.g. `_`, `_extra_`, etc.) and create pure Python module `imgui.extra` that would import those prefixed names and rename them. This will definitely create some issues for Sphinx docs but maybe we'll be able to find some workarounds.

File: imgui/__init__.py [hunk 1]
```diff
@@ -267,3 +269,24 @@
...
+
+
+core.font = _core_to_extra_deprecated(extra.font)
+core.styled = _core_to_extra_deprecated(extra.styled)
+core.istyled = _core_to_extra_deprecated(extra.istyled)
+core.vertex_buffer_vertex_pos_offset = _core_to_extra_deprecated(extra.vertex_buffer_vertex_pos_offset)  # noqa
+core.vertex_buffer_vertex_uv_offset = _core_to_extra_deprecated(extra.vertex_buffer_vertex_uv_offset)  # noqa
+core.vertex_buffer_vertex_col_offset = _core_to_extra_deprecated(extra.vertex_buffer_vertex_col_offset)  # noqa
+core.vertex_buffer_vertex_size = _core_to_extra_deprecated(extra.vertex_buffer_vertex_size)  # noqa
+core.index_buffer_index_size = _core_to_extra_deprecated(extra.index_buffer_index_size)  # noqa
```
Comment (pconerly) [inline imgui/__init__.py]: Random question, what does the `# noqa` comment mean?

**Comment by pconerly** [comment]
LGTM, just a couple of small things. The build is failing, but it's not entirely clear why? Pythonhosted was giving 404s? You might give it a retry.

File: imgui/extra.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+"""
+This module provides extra utilities that are not part of core ImGui C++ API
+but are useful in Python application.
+
+"""
+from . import core
```
[the remark this answers is not in the document]
Comment (swistakm) [reply imgui/extra.py]: `import core` will fail on Python 3 as it no longer supports implicit relative imports.

File: imgui/__init__.py (hunk 1 above)
Comment (swistakm) [reply imgui/__init__.py]: This common mark for linters like `pep8` (and many IDEs) to exclude current line from style checking.  This line is longer than 79 columns but splitting into multiple lines would IMHO reduce readability.

Such things are already my habit. Now I see that project doesn't have linting in build pipeline. This is oversight. I will add another stage to Travis CI config so all `# noqa` will have a bit more sense.

**Comment by swistakm** [comment]
These pip failures are strange indeed. I have restarted Travis build immediately but apparently forgot about Appveyor.

File: setup.py [hunk 3]
```diff
@@ -81,7 +81,21 @@ def get_version(version_tuple):
     general_macros = []
 
 
-extension_sources = ["imgui/core" + ('.pyx' if USE_CYTHON else '.cpp')]
+def extension_sources(path):
+    sources = [path + ('.pyx' if USE_CYTHON else '.cpp')]
```
[the remark this answers is not in the document]
Comment (swistakm) [reply setup.py]: Good point, it would look better.

**Comment by swistakm** [comment]
Apparently this was just a transient issue.

File: imgui/extra.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (pconerly) [reply imgui/extra.py]: ah, TIL!

**Comment by pconerly** [comment]
:+1:

**Comment by swistakm** [comment]
Released as `0.1.0`

**Pull request merged**